packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,354 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { Download, Upload, IZosFilesResponse, IDownloadOptions, IUploadOptions } from "@zowe/zos-files-for-zowe-sdk";
+import { AbstractSession, IHandlerParameters, ImperativeError, ProcessUtils, GuiResult,
+    TextUtils, IDiffNameOptions } from "@zowe/imperative";
+import { CompareBaseHelper } from "../compare/CompareBaseHelper";
+import { CliUtils } from "@zowe/imperative";
+import { existsSync, unlinkSync } from "fs";
+import { tmpdir } from "os";
+import path = require("path");
+import LocalfileDatasetHandler from "../compare/lf-ds/LocalfileDataset.handler";
+import LocalfileUssHandler from "../compare/lf-uss/LocalfileUss.handler";
+
+
+/**
+ * enum of prompts to be used as input to {@link EditUtilities.promptUser} during the file editing process
+ * @export
+ * @enum
+ */
+export enum Prompt {
+    useStash,
+    doneEditing,
+    continueToUpload,
+    viewUpdatedRemote
+}
+
+/**
+ * The possible types of file {@link ILocalFile}
+ * @export
+ * @type
+ */
+export type EditFileType = "uss" | "ds";
+
+/**
+ * A class to hold pertinent information about the local file during the editing process
+ * @export
+ * @interface
+ */
+export interface ILocalFile {
+    tempPath: string | null;
+    fileName: string;
+    fileType: EditFileType;
+    guiAvail: boolean;
+    zosResp: IZosFilesResponse | null;
+    encoding?: string | null;
+}
+
+/**
+ * A shared utility class that uss and ds handlers use for local file editing
+ * @export
+ * @class
+ */
+export class EditUtilities {
+    /**
+     * Builds a temp path where local file will be saved. If uss file, file name will be hashed
+     * to prevent any conflicts with file naming. A given filename will always result in the
+     * same unique file path.
+     * @param {ILocalFile} lfFile - object containing pertinent information about the local file during the editing process
+     * @returns {Promise<string>} - returns unique file path for temp file
+     * @memberof EditUtilities
+     */
+    public static async buildTempPath(lfFile: ILocalFile, commandParameters: IHandlerParameters): Promise<string>{
+        // find the appropriate extension for either uss or ds in one long operation
+        const ext = '.'  + (lfFile.fileType === 'uss'? lfFile.fileName.split(".").pop() : (commandParameters.arguments.extension ?? "txt"));
```

Comment:
If the USS file has no extension, the temp file path generated here will be invalid (at least in Windows) and the file will be inaccessible.

Reply:
Thanks Adam!

Reply:
Still doing something weird:
![image](https://github.com/zowe/zowe-cli/assets/122040687/1f62ac72-5286-4ca7-971e-4afe89d20e0a)